sdk/core/core-client/src/utils.ts

```diff
@@ -37,6 +37,59 @@ export function isValidUuid(uuid: string): boolean {
   return validUuidRegex.test(uuid);
 }
 
+/**
+ * Representation of parsed response headers and body coupled with information
+ * about how to map them:
+ * - whether the response body should be wrapped (typically if its type is primitive).
+ * - whether the response is nullable so it can be null if the combination of
+ *   the headers and the body is empty.
+ */
+interface ResponseObjectWithMetadata {
+  /** whether the mapper allows nullable body */
+  hasNullableType: boolean;
+  /** whether the type of the response body is primitive */
+  shouldWrapBody: boolean;
+  /** parsed headers of the response */
+  headers:
+    | {
+        [key: string]: unknown;
+      }
+    | undefined;
+  /** parsed body of the response */
+  body: any;
+}
+
+/**
+ * Maps the response as follows:
+ * - wraps the response body if needed (typically if its type is primitive).
+ * - returns null if the combination of the headers and the body is empty.
+ * - otherwise, returns the combination of the headers and the body.
+ *
+ * @param responseObject - a representation of the parsed response
+ * @returns the response that will be returned to the user which can be null and/or wrapped
+ *
+ * @internal
+ */
+function handleNullableResponseAndWrappableBody(
+  responseObject: ResponseObjectWithMetadata
+): unknown | null {
+  const combinedHeadersAndBody = {
+    ...responseObject.headers,
+    ...responseObject.body
+  };
+  return responseObject.hasNullableType &&
```

Comment:
nit: would it be easier to read the possible cases if we use `if` checking and return directly from each branch?

Reply:
Yeah, I think 3 ternary operators is a bit much, even though it's pretty straightforward logically speaking

Reply:
Ok I used an outer if but used the simpler form inside each branch. Hopefully this is easier to read.